src/fmeval/eval_algorithms/eval_algorithm.py

```diff
@@ -1,65 +1,49 @@
 from abc import ABC, abstractmethod
-from typing import List, Optional
-from fmeval.data_loaders.util import DataConfig
+from typing import List
 from fmeval.eval_algorithms import EvalScore, EvalOutput
-from fmeval.model_runners.model_runner import ModelRunner
-
-from fmeval.util import get_eval_results_path
 
 
 class EvalAlgorithmConfig:
-    """Config class to be used or extended to provide eval algorithm specific parameters."""
+    """Configuration class to be used or extended to provide evaluation algorithm-specific parameters."""
 
 
 class EvalAlgorithmInterface(ABC):
-    """
-    Interface class for eval algorithms.
-    """
+    """Interface for evaluation algorithms.
 
-    eval_name: str
+    This interface defines two required methods that all evaluation algorithms must implement.
+    The signatures of these methods is intentionally as generic as possible, to allow for
+    maximum freedom when implementing a new evaluation algorithm.
+    """
 
-    def __init__(self, eval_algorithm_config: EvalAlgorithmConfig):
-        """Initialize an instance of a subclass of EvalAlgorithmConfig
+    @abstractmethod
+    def evaluate_sample(self, *args, **kwargs) -> List[EvalScore]:
```

Comment:
why are we moving towards `*args, **kwargs` here? Idea was to let the customers see what are the acceptable arguments for built-in eval algorithms too. 

+ args and kwargs is a slippery slope, opens door for having n number of input variables, leading to confusing signatures. We also want to let these signatures to serve as guidelines when contributing new eval algorithms, not leave them completely open ended for any input variable. 

We want to keep them generic, agreed but not give complete freedom to take in any input variable.

Reply:
Discussed offline and I will restore the original signatures in a follow-up PR.